plugins/module_utils/redfish_utils.py

```diff
@@ -2311,11 +2311,17 @@ def set_bios_attributes(self, attributes):
 
         # Construct payload and issue PATCH command
         payload = {"Attributes": attrs_to_patch}
+        additional_data = ''
+
+        if 'PwrButton' in attrs_to_patch:
+            additional_data = '. Reboot required to apply setting'
+            payload.update({"@Redfish.SettingsApplyTime": {"ApplyTime": "OnReset"}})
+
         response = self.patch_request(self.root_uri + set_bios_attr_uri, payload)
         if response['ret'] is False:
             return response
         return {'ret': True, 'changed': True,
-                'msg': "Modified BIOS attributes %s" % (attrs_to_patch),
+                'msg': "Modified BIOS attributes %s%s" % (attrs_to_patch, additional_data),
```

Comment:
Is there a need for "additional data"? Everyone requires a reboot for the settings to be applied (but everyone other than Dell tracks this internally on behalf of the user).

Reply:
That's a good point; it might be worth just changing the message returned to always state "a reboot is required".

Reply:
Good suggestion. I've pushed it.